libsolidity/analysis/ViewPureChecker.cpp

```diff
@@ -179,7 +179,7 @@ void ViewPureChecker::endVisit(Identifier const& _identifier)
 	bool writes = _identifier.annotation().lValueRequested;
 	if (VariableDeclaration const* varDecl = dynamic_cast<VariableDeclaration const*>(declaration))
 	{
-		if (varDecl->isStateVariable())
+		if (varDecl->isStateVariable() && !varDecl->isConstant())
```

Comment:
Unrelated to this PR: I wonder if a constant should be considered a state variable?

Reply:
Yes, we should probably change that helper function.